### PR TITLE
Chameleon kit tweaks

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

Added blacklist to chameleon items, currently it contains most bad_type's and BST uniform.
Chameleon verbs have been moved to "Object" tab, "Chameleon Items" no longer exist.
Chameleon kit comes with only one backpack.
Not only it comes with one backpack, it comes _**in**_ a backpack instead of a box.
Chameleon gun now properly mimicking both energy and ballistic projectiles.
Chameleon gun spawns with no serials, but those could be set (and removed) via verb.

https://user-images.githubusercontent.com/65828539/163893947-7020f815-24ba-4ca3-9605-d0248afa25fd.mp4

https://user-images.githubusercontent.com/65828539/163894530-7e48eae5-22ac-4e62-8d76-87fc678802b9.mp4

## Why It's Good For The Game

Features and fixes is cool, duplicated code and broken stuff is uncool.

## Changelog
:cl:
add: chameleon gun can change it's own serial number
del: chameleon satchel
tweak: chameleon kit comes in a chameleon backpack
tweak: chameleon and handmade guns comes with no serial number
fix: chameleon projectiles not changing appearance properly
refactor: chameleon items code
/:cl:

### CODE REVIEW

**Comment by Alliostra** [comment]
The 2 "backpacks" were intended so you can choose between using a satchel or a backpack. If you only have the backpack, and you disguise it as a satchel you need to always take it off your back to access it, which makes you look sus. 
Most people prefer using a satchel over a backpack anyways, so if you really want to remove either, it's better to remove the backpack instead.

Also I'm not sure if moving all the verbs to the objects tab is a good idea for accessibility, since in that tab the position of the verbs often change very quickly. Adding HUD action buttons (like the ones for activating flashlights and so on) might be a good idea?

File: code/modules/acting/acting_items.dm [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	if(active)
 		active = 0
 		spawn(30)
-			new /obj/item/storage/box/syndie_kit/chameleon(src.loc)
+			new /obj/item/storage/backpack/chameleon(src.loc)
 			src.visible_message("\The [src] beeps, dispensing a small box onto the floor.", "You hear a beeping sound followed by a thumping noise of some kind.")
```
Comment (hyperioo) [inline code/modules/acting/acting_items.dm:19]:
```suggestion
			new /obj/item/storage/backpack/chameleon(loc)
			src.visible_message("\The [src] beeps, dispensing a backpack onto the floor.", "You hear a beeping sound followed by a thumping noise of some kind.")
```

File: code/modules/clothing/chameleon.dm [hunk 2]
```diff
@@ -17,29 +78,21 @@
...
 
-	return copy //for inheritance
-obj/item/clothing/disguise(newtype, mob/user)
-	. = ..()
-	var/obj/item/clothing/copy = .
-	if (istype(copy))
-		style_coverage = copy.style_coverage
+	if(istype(copy, /obj/item/clothing))
+		var/obj/item/clothing/C = copy
+		src:style_coverage = C.style_coverage
```
Comment (hyperioo) [inline code/modules/clothing/chameleon.dm:86]:
```suggestion
		style_coverage = C.style_coverage
```

**Comment by hyperioo** [comment]
Would be great to have a smol separate PR for the handgun serial number removal.

File: code/modules/clothing/chameleon.dm (hunk 2 above)
Comment (SirRichardFrancis) [reply code/modules/clothing/chameleon.dm:86]: That will not compile.

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -97,7 +97,7 @@
 	var/inversed_carry = FALSE
 	var/wield_delay = 0 // Gun wielding delay , generally in seconds.
 	var/wield_delay_factor = 0 // A factor that characterizes weapon size , this makes it require more vig to insta-wield this weapon or less , values below 0 reduce the vig needed and above 1 increase it
-	var/serial_type = "SM" // Self-manufactured , if there is a serial type , the gun will add a number onto its final , if none , it won;'t show on examine
+	var/serial_type = "" // If there is a serial type, the gun will add a number onto its final, if none, it won't show on examine
```
Comment (SirRichardFrancis) [inline code/modules/projectiles/gun.dm:100]:
```suggestion
var/serial_type = "SM" // Self-manufactured , if there is a serial type , the gun will add a number onto its final , if none , it won;'t show on examine
```

File: code/modules/projectiles/gun.dm [hunk 4]
```diff
@@ -97,7 +97,7 @@
 	var/inversed_carry = FALSE
 	var/wield_delay = 0 // Gun wielding delay , generally in seconds.
 	var/wield_delay_factor = 0 // A factor that characterizes weapon size , this makes it require more vig to insta-wield this weapon or less , values below 0 reduce the vig needed and above 1 increase it
-	var/serial_type = "SM" // Self-manufactured , if there is a serial type , the gun will add a number onto its final , if none , it won;'t show on examine
+var/serial_type = "SM" // Self-manufactured , if there is a serial type , the gun will add a number onto its final , if none , it won;'t show on examine
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:100]:
```suggestion
	var/serial_type = "SM" // Self-manufactured , if there is a serial type , the gun will add a number onto its final , if none , it won;'t show on examine
```
You removed the indent

**Comment by SirRichardFrancis** [comment]
Wtf?
image

**Comment by hyperioo** [comment]
It's likely from #7191 I warned the author on Discord

**Comment by hyperioo** [comment]
Merging, we'll have to see if having the verbs in Object tab is painfull (in that case HUD button could be a nice way). Merging because integration test were working before the other mech PR, the fail is unrelated.

**Pull request merged**